### PR TITLE
Infra: fix typos in initial set up

### PR DESCRIPTION
**Pull request by skipcloud**
A few fixes for the initial terraform which I wasn't able to plan until we got geop working

Ticket number NEC-2356
https://deliveroo.atlassian.net/browse/NEC-2356

### CODE REVIEW

File: infrastructure/modules/basic/iam.tf [hunk 1]
```diff
@@ -20,7 +20,39 @@ resource "aws_iam_role" "circleci_oidc" {
   assume_role_policy = data.aws_iam_policy_document.assume_policy_circleci_oidc.json
 }
 
+data "aws_iam_policy" "circleci_oidc" {
+  name = "CircleCI-OIDC-Access"
```
Comment (fergalhk) [inline infrastructure/modules/basic/iam.tf:24]: I'm not sure that this policy will give us what we need unfortunately - it's scoped to just `ecr:` actions, whereas we need `ecr-public:` actions 😞

Comment (skipcloud) [reply infrastructure/modules/basic/iam.tf:24]: Oh yeah you're right, I even made the right changes below in the policy hah. Let me port over the policy into this repo so we can alter it.

File: infrastructure/modules/basic/iam.tf [hunk 2]
```diff
@@ -20,7 +20,60 @@ resource "aws_iam_role" "circleci_oidc" {
...
+      "ecr-public:DescribeRepositories",
+      "ecr-public:GetDownloadUrlForLayer",
+      "ecr-public:InitiateLayerUpload",
+      "ecr-public:ListImages",
+      "ecr-public:PutImage",
+      "ecr-public:UploadLayerPart",
+    ]
+
+    resources = [
+      "arn:aws:ecr:us-east-1:${data.aws_caller_identity.current.account_id}:repository/&{aws:PrincipalTag/ECRRepoName}*",
```
Comment (fergalhk) [inline infrastructure/modules/basic/iam.tf:38]: I'm not following this ARN - do you think it's worth using `aws_ecrpublic_repository.nginx-sidecar.arn` instead?

Comment (skipcloud) [reply infrastructure/modules/basic/iam.tf]: you're right, this is a hold over from the shared IAM policy I copy/pasta'd this from

**Review by fergalhk — APPROVED** [review]

**Pull request merged**